example/fine_advection/advection_driver.cpp

```diff
@@ -0,0 +1,165 @@
+  pin->CheckDesired("Advection", "cfl");
+  pin->CheckDesired("Advection", "vx");
+  pin->CheckDesired("Advection", "refine_tol");
+  pin->CheckDesired("Advection", "derefine_tol");
+}
+
+// See the advection.hpp declaration for a description of how this function gets called.
+TaskCollection AdvectionDriver::MakeTaskCollection(BlockList_t &blocks, const int stage) {
+  using namespace parthenon::Update;
+  TaskCollection tc;
+  TaskID none(0);
+
+  // Build MeshBlockData containers that will be included in MeshData containers. It is
+  // gross that this has to be done by hand.
+  const auto &stage_name = integrator->stage_name;
+  if (stage == 1) {
+    for (int i = 0; i < blocks.size(); i++) {
+      auto &pmb = blocks[i];
+      // first make other useful containers
+      auto &base = pmb->meshblock_data.Get();
+      pmb->meshblock_data.Add("dUdt", base);
+      for (int s = 1; s < integrator->nstages; s++)
+        pmb->meshblock_data.Add(stage_name[s], base);
+    }
+  }
+
+  const Real beta = integrator->beta[stage - 1];
+  const Real dt = integrator->dt;
+
+  const int num_partitions = pmesh->DefaultNumPartitions();
+  TaskRegion &single_tasklist_per_pack_region = tc.AddRegion(num_partitions);
+
+  for (int i = 0; i < num_partitions; i++) {
+    auto &tl = single_tasklist_per_pack_region[i];
+    auto &mbase = pmesh->mesh_data.GetOrAdd("base", i);
+    auto &mc0 = pmesh->mesh_data.GetOrAdd(stage_name[stage - 1], i);
+    auto &mc1 = pmesh->mesh_data.GetOrAdd(stage_name[stage], i);
+    auto &mdudt = pmesh->mesh_data.GetOrAdd("dUdt", i);
+
+    auto start_send = tl.AddTask(none, parthenon::StartReceiveBoundaryBuffers, mc1);
+    auto start_flxcor = tl.AddTask(none, parthenon::StartReceiveFluxCorrections, mc0);
+
+    static auto desc =
+        parthenon::MakePackDescriptor<advection_package::Conserved::scalar>(
+            pmesh->resolved_packages.get(), {parthenon::Metadata::WithFluxes},
+            {parthenon::PDOpt::WithFluxes});
+    using pack_desc_t = decltype(desc);
+
+    static auto desc_fine =
+        parthenon::MakePackDescriptor<advection_package::Conserved::scalar_fine>(
+            pmesh->resolved_packages.get(), {parthenon::Metadata::WithFluxes},
+            {parthenon::PDOpt::WithFluxes});
+    using pack_desc_fine_t = decltype(desc_fine);
+
+    using TE = parthenon::TopologicalElement;
+    std::vector<TE> faces{TE::F1};
+    if (pmesh->ndim > 1) faces.push_back(TE::F2);
+    if (pmesh->ndim > 2) faces.push_back(TE::F3);
+    auto flx = none;
+    auto flx_fine = none;
+    for (auto face : faces) {
+      flx = flx | tl.AddTask(none, advection_package::CalculateFluxes<pack_desc_t>, desc,
+                             face, parthenon::CellLevel::same, mc0.get());
+      flx_fine = flx_fine |
+                 tl.AddTask(none, advection_package::CalculateFluxes<pack_desc_fine_t>,
+                            desc_fine, face, parthenon::CellLevel::fine, mc0.get());
+    }
```

Comment:
Ah I see. Cute. Would it not be better maybe though to just use the var names for the template params rather than the pack descriptor type? I guess it doesn't matter...